src/core/dataProvider/dataprovider-archivedb/block-queries-sql.ts

```diff
@@ -103,6 +103,26 @@ async function getNullParents(minHeight: number, maxHeight: number) {
     return heights.map((x) => x.height);
 }
 
+export async function getMinBlockHeightBySlots(minSlot: number, maxSlot: number) {
+    const result = await db.one<height>(`
+    select height from blocks where global_slot between $1 and $2
```

Comment:
Because not considering canonical blocks only, this ends up with a different range than expected. Should only consider blocks in the canonical chain. (Same with Max)

Reply:
Test run for -e=14
ME: Epoch Minimum Height: 69767 - Epoch Maximum Height: 74463
Archive: Epoch Minimum Height: 69594 - Epoch Maximum Height: 74463

Reply:
Not plugging this in directly in case you disagree with changing the signature, but this also eliminates the cost of the order by, but of course incurs the much more expensive recursive.... which I think is one more reason in favor of a single query to get both min and max ... perhaps consider changing ME impl to also get min/max in one call.

SELECT min(height) as minHeight, max(height) as maxHeight from blocks b 
WHERE id IN 
(
WITH RECURSIVE chain AS 
	(
	SELECT id, parent_id FROM blocks b WHERE height = (select MAX(height) from blocks)
	
	UNION ALL

	SELECT b.id, b.parent_id FROM blocks b 
	INNER JOIN chain
	ON b.id = chain.parent_id
	) 

	SELECT distinct c.id
	FROM chain c
)
and global_slot between $1 and $2